### DISCUSSION TITLE
Making a V 1.20 micropython binary for thonny to "install"

### DISCUSSION THREAD

**Discussion by liudr** [micropython/micropython#11841] [ESP32]
I am able to do make [board] deploy to flash new build onto a new board. I'd like to be able to do that without invoking the make but rather just use thonny to install like I was able to do it with official releases or using esptool to flash the binary.

I tried to have thonny install my compiled binary micropython.bin but the outcome was the "app" failed to boot or other errors. I wonder if it is due to the exact procedure thonny uses to install the firmware instead of something wrong with my compiled firmware. I know that I could flash the firmware together with bootloader and partition table using esptool but for others to use my firmware it's easier to use thonny. Maybe thonny has a set bootloader and partition binary that has worked on earlier versions of mp but not with 1.20? Thanks.

**Comment by jimmo** [comment]
@liudr Sorry I'm not quite sure I understand your question. This may be better raised as an issue against Thonny? MicroPython just builds firmware images that should work with any flashing tool.

**Comment by shariltumin** [comment]
Now that you have managed to build (i.e. make V=1 BOARD=XXXX), you will have a build-XXXX directory. There will be a file called **firmware.bin** in this directory. Install **firmware.bin** using thonny. You tried to install the wrong file. The *micropython.bin* does not contain the boot loader and partition table.

**Comment by shariltumin** [comment]
Oh, before I forget. Thonny was probably set up to flash ESP32 only (at least the one I have, version 4.0.1). It will probably fail to flash ESP32S3 since flash_mode and location are different than ESP32.
